provers/evm-prover/ARCHITECTURE.md

```diff
@@ -0,0 +1,76 @@
+# EVM Prover
+
+The EVM Prover implements the Prover service for EVM rollups, creating zero-knowledge range proofs for state membership and state transitions. It uses the following SP1 programs to generate these range proofs.
+
+## SP1 Programs
+
+### blevm
+
+The `blevm` SP1 program generates proofs of data inclusion and execution for single EVM blocks.
```

Comment:
[nit] in a follow-up can this program be renamed? I still don't understand the rationale behind the original name (maybe blob EVM)? A few ideas for the new program:
- single-evm-block
- evm-block